DigitalLearningSolutions.Web/Views/Shared/Components/DateRangeInput/_DateInput.cshtml

```diff
@@ -0,0 +1,55 @@
+﻿@using DigitalLearningSolutions.Web.ViewModels.Common.ViewComponents
+@model DateRangeInputDateInputViewModel
+
+@{
+  var dayErrorCss = Model.HasDayError ? "nhsuk-input--error" : "";
+  var monthErrorCss = Model.HasMonthError ? "nhsuk-input--error" : "";
+  var yearErrorCss = Model.HasYearError ? "nhsuk-input--error" : "";
+  var conditionalInputClass = Model.IsConditionalDateInput ? "nhsuk-checkboxes__conditional" : "";
+  var conditionalInputId = Model.IsConditionalDateInput ? "conditional-end-date" : "start-date";
+  var isHiddenClass = Model.IsHidden ? "nhsuk-checkboxes__conditional--hidden" : "";

```

Comment:
Shouldn't this be merged with L8 as it's either a `conditional` - or `conditional--hidden`?

Reply:
No, there are 2 different states we need to handle:

- Is conditional & Is Hidden - we want both classes
- Is conditional & Is not Hidden - we only want the non --hidden class
- Is not conditional & Is Hidden - invalid state
- Is not conditional & is not hidden - We want neither class (used for the normal non-hidden input start date)